pkg/metrics/metrics.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate protoc --go_out=. --go_opt=paths=source_relative clientanalytics.proto
+//go:generate protoc --go_out=. --go_opt=paths=source_relative internal_user_log.proto
+package metrics
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/uuid"
+)
+
+const (
+	ClientTypeValue    = 16
+	LogSourceValue     = 971
+	LogSourceNameValue = "CUTTLEFISH_METRICS"
+	ToolName           = "cvdr"
+)
+
+var (
+	UserTypeVal    = UserType_GOOGLE
+	TestReferences = []string{"cvdr"}
+)
+
+func currentTimeMillis() int64 {
+	return time.Now().UnixNano() / int64(time.Millisecond)
+}
+
+func encodeLogRequest(extension []byte) ([]byte, error) {
+	currentTimeMs := currentTimeMillis()
+	log.Printf("current time = %v", currentTimeMs)
```

Comment:
Is this a leftover from debugging? It doesn't seem to have much value since logs typically include the date.

Reply:
Done